core/backend/src/CheckpointManager.ts

```diff
@@ -330,6 +330,7 @@ export class CheckpointManager {
           nativeDb.resetBriefcaseId(BriefcaseIdValue.Unassigned);
 
         CheckpointManager.validateCheckpointGuids(checkpoint, nativeDb);
+        (db as any).iModelId = nativeDb.getIModelId();
```

Comment:
Does anyone have better suggestions for how to fix this? We could look at the nativeDb in BriefcaseManager.pullAndApplyChangesets when we're grabbing the iModelId... but then we still have an outdated iModelId stored on the TS object. We can't update it in validateCheckpointGuids because it takes a DgnDb.

Reply:
i don't understand. Is this a V2 checkpoint?

Reply:
Yes its a V2 checkpoint. Initially, I thought we couldn't do that since its a public function... but I guess since its internal we wouldn't be in violation of anything right?

Reply:
yes, internal functions can be changed. But how did we get an invalid V2 checkpoint? That should be where we fix this problem, right?

Reply:
Its quite an old iModel used for testing that was made in 2019 I believe. At that time, I don't think the hub was resetting iModelId when copying iModels

Reply:
You're right, I'll make that change in the v2 checkpoint agent.

Reply:
what "agent" this this come up in?

Reply:
changed elements agent